spec/models/asset_spec.rb

```diff
@@ -284,9 +284,11 @@
     end
 
     it "cannot be changed after creation" do
-      uuid = "11111111-1111-1111-1111-11111111111111"
-      asset = FactoryBot.create(:asset, uuid: uuid)
-      expect { asset.uuid = "22222222-2222-2222-2222-222222222222" }.to raise_error(Mongoid::Errors::ReadonlyAttribute)
```

Comment:
As of Mongoid 7, attempting to modify a read-only parameter raises an exception only if done via `update_attribute`, but not if done via the setter.

So `asset.update_attribute(:uuid, "222...")` would raise a `Mongoid::Errors::ReadonlyAttribute`, but `asset.uuid = "222..." does not.

The test could be easily modified to check for the error-raising behaviour, if that were functionally important here; but I'm assuming it is not.